trieste/acquisition/function.py

```diff
@@ -777,24 +778,41 @@ def __init__(
         search_space: SearchSpace,
         num_samples: int = 500,
         penalizer: Callable[..., PenalizationFunction] = None,
+        base_acquisition_function_builder: Optional[SingleModelAcquisitionBuilder] = None,
     ):
         """
         :param search_space: The global search space over which the optimisation is defined.
         :param num_samples: Size of the random sample over which the Lipschitz constant
             is estimated. We recommend scaling this with search space dimension.
+        :param penalizer: The chosen penalization method (defaults to soft penalization).
+        :param base_acquisition_function_builder: Base acquisition function to be
+            penalized (defaults to expected improvement).
+
         """
         self._search_space = search_space
         if num_samples <= 0:
             raise ValueError(f"num_samples must be positive, got {num_samples}")
         self._num_samples = num_samples
 
-        if penalizer is None:
-            penalizer = soft_local_penalizer
+        self._lipschitz_penalizer = soft_local_penalizer if penalizer is None else penalizer
 
-        self._lipschitz_penalizer = penalizer
+        if base_acquisition_function_builder is None:
+            self._base_builder = ExpectedImprovement()
```

Comment:
You'll need a type annotation, so mypy doesn't wrongly assume that _base_builder is always an ExpectedImprovement builder.

```suggestion
            self._base_builder: SingleModelAcquisitionBuilder = ExpectedImprovement()
```

Alternatively you could rewrite/simplify it:

```suggestion
            self._base_builder = base_acquisition_function_builder or ExpectedImprovement()
            if not isinstance(self._base_builder, (ExpectedImprovement, MinValueEntropySearch)):
                ....
```